### PR TITLE
Embed API: Glossary terms sharing description (Sphinx)

### PR DESCRIPTION
**Pull request by humitos**
I wasn't able to update https://github.com/readthedocs/readthedocs.org/pull/9984 by pushing my changes, so I went ahead and created a new PR for it that includes my changes as well 🤷🏼 cc @jpfeuffer

Closes https://github.com/readthedocs/sphinx-hoverxref/issues/97

### CODE REVIEW

File: readthedocs/embed/v3/views.py [hunk 1]
```diff
@@ -214,8 +214,18 @@ def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversio
...
+                        traverse = node.traverse()
+                        iteration = 0
+                        while iteration < 5:
+                            next_node = next(traverse, None)
+                            # TODO: Do we need to support terms with missing descriptions?
+                            # This will not produce correct results in this case.
+
+                            # Stop at the next 'dd' node, which is the description
+                            if iteration >= 5 or (next_node and next_node.tag == "dd"):
+                                iteration = 99  # stop the iteration
```
Comment (benjaoming) [inline readthedocs/embed/v3/views.py:228]: why not `break`?

Comment (humitos) [reply readthedocs/embed/v3/views.py:228]: Just because I'm dumb 😝 . I fixed this. Thanks!

**Comment by jpfeuffer** [comment]
Awesome. This will make our docs much nicer! Thank you. I don't care about the closed PR of course.

**Comment by humitos** [comment]
I fixed the tests. This is ready to merge once I get an approve.

File: readthedocs/embed/v3/views.py [hunk 2]
```diff
@@ -214,8 +214,19 @@ def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversio
...
+                        traverse = node.traverse()
+                        iteration = 0
+                        while iteration < 5:
+                            next_node = next(traverse, None)
+                            # TODO: Do we need to support terms with missing descriptions?
+                            # This will not produce correct results in this case.
+
+                            # Stop at the next 'dd' node, which is the description
+                            if iteration >= 5 or (next_node and next_node.tag == "dd"):
+                                break
```
Comment (ericholscher) [inline readthedocs/embed/v3/views.py:228]: Isn't the `iteration >= 5` already provided by the `while iteration < 5`?

File: readthedocs/embed/v3/views.py [hunk 3]
```diff
@@ -214,8 +214,19 @@ def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversio
...
                     parent_node = node.parent
-                    if 'glossary' in node.parent.attributes.get('class'):
-                        next_node = node.next
+                    if "glossary" in node.parent.attributes.get("class"):
+                        # iterate through child and next nodes
+                        traverse = node.traverse()
+                        iteration = 0
+                        while iteration < 5:
+                            next_node = next(traverse, None)
+                            # TODO: Do we need to support terms with missing descriptions?
```
Comment (ericholscher) [inline readthedocs/embed/v3/views.py:223]: Should we raise an error here? What is "not produce correct results"?

**Review by ericholscher — APPROVED** [review]
Seems fine?

File: readthedocs/embed/v3/views.py (hunk 2 above)
Comment (benjaoming) [reply readthedocs/embed/v3/views.py:228]: indeed, the first part of the condition can be removed, but it's also harmless.

File: readthedocs/embed/v3/views.py (hunk 3 above)
Comment (humitos) [reply readthedocs/embed/v3/views.py:223]: I haven't tested this case. However, I think it's not possible to create a glossary entry without a description, tho

**Pull request merged**